python/grass/jupyter/init_funcs.py

```diff
@@ -0,0 +1,61 @@
+import os
+
+import grass.script as gs
+import grass.script.setup as gsetup
+
+
+def _set_notebook_defaults():
+    """
+    This function sets several GRASS environment variables that are
+    important for GRASS to run smoothly in Jupyter. A complete list and
+    description of environment variables can be found here:
+    https://grass.osgeo.org/grass78/manuals/variables.html
+    """
+    # We want functions to raise exceptions and see standard output of
+    # the modules in the notebook.
+    gs.set_raise_on_error(True)
+    gs.set_capture_stderr(True)
+
+    # Allow overwrite of existing maps
+    os.environ["GRASS_OVERWRITE"] = "1"
+
+
+def init(grassdata_filepath, location, mapset):
```

Comment:
`grassdata_filepath` is too long and `filepath` part is not great. You can try:

```
init(path, location=None, mapse=None):
```

`path` will be path to mapset or path to database if location and mapset were provided. To follow the new v8 GUI behavior of using PERMANENT mapset automatically, you could even allow providing only `path` and `location`, but not mapset and also if only `path` is provided and points to an existing location (`grass.grassdb` is your friend), assume mapset=PERMANENT.

Reply:
I'm not quite sure what this would look like (or that I understand correctly). Maybe like this:
`def init(path, location=None):
.
.
.
gsetup.init(os.environ["GISBASE"], path, location, mapset="PERMANENT")`

Is that close to what you were thinking? If location is None, I think this would cause an error?

Reply:
Let's just leave this for another PR, so that you get the basic things working quickly and can use them. It's not just about setting some combination of defaults, but rather about running series of checks to figure out what the user meant.